### PR TITLE
Remove var names

### PR DESCRIPTION
**Pull request by cgsandford**
Description of changes:

- Added line to strip var names on load using utility that retains the probabilistic `var_name="threshold"` coordinate
- Removed all cases where `var_name` is explicitly set in coordinate creation, except for the percentile coordinate created during ECC which is addressed in #1211 
- Removed two over-generalised functions, `build_coordinate` and `add_coord`, which were unnecessarily manipulating the `var_name`, and replaced calls to both functions with simpler alternatives

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ improver/metadata/amend.py  -8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5336237&bid=17467222)

File: improver_tests/blending/blend_across_adjacent_points/test_TriangularWeightedBlendAcrossAdjacentPoints.py [hunk 1]
```diff
@@ -232,24 +229,23 @@ def test_works_one_thresh(self):
         # Take a slice of the time coordinate.
         expected_cube = self.cube[0].copy(data.astype(np.float32))
 
-        # Add threshold axis to expected output cube.
+        # Add height axis to expected output cube.
         changes = {'points': [0.5], 'units': '1', 'var_name': 'threshold'}
```
Comment (fionaRust) [inline improver_tests/blending/blend_across_adjacent_points/test_TriangularWeightedBlendAcrossAdjacentPoints.py:233]: Need to get rid of this dictionary too?

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ improver/metadata/amend.py  -8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5336237&bid=17467222)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity decreasing per file
==============================
+ improver/metadata/amend.py  -8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5336237&bid=17467222)

File: improver_tests/utilities/test_load.py [hunk 2]
```diff
@@ -229,6 +229,25 @@ def test_none_file_without_allow_none(self):
...
+        self.assertIsNone(result.coord("realization").var_name)
+
+    def test_threshold_var_name_retained(self):
+        """Test necessary coordinate var name is retained"""
+        cube = set_up_probability_cube(
+            np.ones((3, 3, 3), dtype=np.float32),
+            np.array([285, 286, 287], dtype=np.float32))
+        os.remove(self.filepath)
+        save_netcdf(cube, self.filepath)
+        result = load_cube(self.filepath)
```
Comment (benfitzpatrick) [inline improver_tests/utilities/test_load.py:248]: Codacy Issue found: [Unused variable 'result'](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5336237)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ improver/metadata/amend.py  -8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5336237&bid=17467222)

**Pull request merged**